tnoodle-ui/src/components/EventsTable.jsx

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from "react";
+import CubingIcon from "../CubingIcon";
+
+class EventsTable extends Component {
+  render() {
+    const { events } = this.props;
+    let maxRounds = Math.max.apply(
+      Math,
+      events.map(event => event.rounds.length)
+    );
+    return (
+      <div className="text-center">
+        <table className="table table-striped table-bordered table-hover table-condensed">
+          <thead className="thead-dark">
+            <tr>
+              <th scope="col" />
+              <th scope="col">Event</th>
+              {Array.from(Array(maxRounds), (e, i) => {
+                return (
+                  <th key={i} scope="col">
+                    Groups for Round {i + 1}
+                  </th>
+                );
+              })}
+            </tr>
+          </thead>
+          <tbody>
+            {events.map(event => (
+              <tr>
```

Comment:
Does this `tr` need a `key` attribute? I'd expect you to see warnings in the console if you don't have one.

Reply:
Fixed